documentation/optee_design.md

```diff
@@ -487,6 +487,88 @@ implementation, as explained in [crypto.md].
 TBD
 
 # 11. Trusted Applications
+
+## Static and Dynamic TAs
+
+There are two ways to implement Trusted Applications, static and dynamic.
+For most cases, Dynamically-loaded TAs are preferred.
+
+### Static Trusted Applications
+
+These are added directly to the optee_os tree in, eg, `core/arch/arm/sta`, and
+are built along with and built into the OS blob.
+
+You can look at `core/arch/arm/sta/stats.c` as a template and just add your
+static TA based on that to the `sub.mk` in the same directory.
+
+Static TAs have the same privileged view of Secure World as the optee_os core
+itself.  For complex TAs, that may not be desirable.
+
+### Dynamic Trusted Applications
+
+Trusted Applications consists of a signed ELF file.
```

Comment:
Also rather change "Dynamic Trusted Applications" into "Trusted Applications" and start with this:
```
### Trusted Applications

Trusted Applications are applications loaded by OP-TEE core in the CPU userland unpriviledge world in secure state. Trusted Application benefit from the `GP Core Internal API` as specified by the 'GP Specs'. 

Trusted Applications (TAs) are also called dynamic TAs since the OP-TEE creates at runtime an execution environment for the TAs including their loading from the REE filesystem.

Trusted Application consists of a cleartext signed ELF file.